### PR TITLE
KYLIN-4612 support job status writing to kafka

### PR DESCRIPTION
**Pull request by bigxiaochu**

### CODE REVIEW

**Comment by hit-lacus** [comment]
Could youp please try to check this documenation : https://cwiki.apache.org/confluence/display/KYLIN/2020/05/14/Collect+and+analyse+Kylin+runtime+with+Real-time+OLAP ?

**Comment by bigxiaochu** [comment]
@hit-lacus 我看了你提供的wiki，这个功能很好，输出metrics到kafka，再kylin自己消费掉生成实时系统cube。我后面会用到这个功能。非常感谢

我这个是输出作业的状态变更到kafka，是给kylin用户消费用的。代替轮训作业列表，因为有上千个作业再运行的话，同时轮训的代价太大了。

可以考虑共用一套写入kafka逻辑，但由于目的不一样，topic不一样，甚至由于公司有多套kafka集群，用的kafka集群可能也不一样，所以最好不要共用一套写入kafka逻辑，现有的KafkaReservoirReporter也是专门处理mertics的。

**Comment by hit-lacus** [comment]
Thank you for your reply, let us rethink how can we do it in better way.

**Comment by hit-lacus** [comment]
In my side, I think this feature is meaningful. I expect you to provided documentation from your team, may included:
1. In which usage scenario do we need this feature
2. How to configure, how to set up
3. What is the benefit, what problem do we solve
4. From dev design, try explain the difference between system cube with your feature.

**Comment by bigxiaochu** [comment]
@hit-lacus 更新了jira描述， 中英文markdown说明

File: core-common/src/main/java/org/apache/kylin/common/kafka/KafkaMsgProducer.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        if (null == TOPIC_NAME) {
+            TOPIC_NAME = kafkaConfig.get("topic.name");
+        }
+    }
+
+    public void sendJobStatusMessage(String message) {
+        sendMessage(message);
+    }
+
+    public void sendEventMessage(String message) {
```
Comment (hit-lacus) [inline core-common/src/main/java/org/apache/kylin/common/kafka/KafkaMsgProducer.java:84]: Is this method useless ?

**Comment by hit-lacus** [comment]
LGTM

**Comment by hit-lacus** [comment]
LGTM

**Pull request merged**